backend_new/src/services/application.service.ts

```diff
@@ -195,68 +211,343 @@ export class ApplicationService {
+              },
+            }
+          : undefined,
+        accessibility: dto.accessibility
+          ? {
+              create: {
+                ...dto.accessibility,
+              },
+            }
+          : undefined,
+        alternateContact: dto.alternateContact
+          ? {
+              create: {
+                ...dto.alternateContact,
+                address: {
+                  create: {
+                    ...dto.alternateContact.address,
+                  },
+                },
+              },
+            }
+          : undefined,
+        applicationsAlternateAddress: dto.applicationsAlternateAddress
+          ? {
+              create: {
+                ...dto.applicationsAlternateAddress,
+              },
+            }
+          : undefined,
+        applicationsMailingAddress: dto.applicationsMailingAddress
+          ? {
+              create: {
+                ...dto.applicationsMailingAddress,
+              },
+            }
+          : undefined,
+        listings: dto.listings
+          ? {
+              connect: {
+                id: dto.listings.id,
+              },
+            }
+          : undefined,
+        demographics: dto.demographics
+          ? {
+              create: {
+                ...dto.demographics,
+              },
+            }
+          : undefined,
+        preferredUnitTypes: dto.preferredUnitTypes
+          ? {
+              connect: dto.preferredUnitTypes.map((unitType) => ({
+                id: unitType.id,
+              })),
+            }
+          : undefined,
+        householdMember: dto.householdMember
+          ? {
+              create: dto.householdMember.map((member) => ({
+                orderId: member.orderId,
+                firstName: member.firstName,
+                middleName: member.middleName,
+                lastName: member.lastName,
+                birthMonth: member.birthMonth,
+                birthDay: member.birthDay,
+                birthYear: member.birthYear,
+                sameAddress: member.sameAddress,
+                relationship: member.relationship,
+                workInRegion: member.workInRegion,
+                householdMemberAddress: {
+                  create: {
+                    ...member.householdMemberAddress,
+                  },
+                },
+                householdMemberWorkAddress: {
+                  create: {
+                    ...member.householdMemberWorkAddress,
+                  },
+                },
+              })),
+            }
+          : undefined,
+        programs: JSON.stringify(dto.programs),
+        preferences: JSON.stringify(dto.preferences),
+      },
+    });
+
+    await this.updateListingApplicationEditTimestamp(res.listingId);
+    return mapTo(Application, res);
   }
 
   /*
-    this will update an application
-    if no application has the id of the incoming argument an error is thrown
+    this will delete an application
```

Comment:
This doesn't actually delete the application, just adds a deletedAt field. We should probably update this comment to be something like "Mark application as deleted"

Reply:
updated comment to be more clear